jerry-core/ecma/operations/ecma-reference.c

```diff
@@ -182,6 +175,91 @@ ecma_op_is_prop_unscopable (ecma_object_t *lex_env_p, /**< lexical environment *
 } /* ecma_op_is_prop_unscopable */
 #endif /* ENABLED (JERRY_ES2015) */
 
+/**
+ * Helper method for HasBindig operation
+ *
+ * See also:
+ *         ECMA-262 v6, 8.1.1.2.1 steps 7-9;
+ *
+ * @return ECMA_VALUE_TRUE - if the property is unscopable
+ *         ECMA_VALUE_FALSE - if a the property is not unscopable
+ *         ECMA_VALUE_ERROR - otherwise
+ */
+
+/**
+ * Resolve value corresponding to the given object environment reference.
+ *
+ * Note: the steps are already include the HasBindig operation steps
+ *
+ *  See also:
+ *         ECMA-262 v6, 8.1.1.2.1
+ *
+ * @return ECMA_VALUE_ERROR - if the operation fails
+ *         ECMA_VALUE_NOT_FOUND - if the binding not exists or blocked via @@unscopables
+ *         result of the binding - otherwise
+ */
+ecma_value_t
+ecma_op_object_bound_environment_resolve_reference_value (ecma_object_t *lex_env_p, /**< lexical environment */
+                                                          ecma_string_t *name_p) /**< variable name */
+{
+  ecma_object_t *binding_obj_p = ecma_get_lex_env_binding_object (lex_env_p);
+  ecma_value_t found_binding;
+
+#if ENABLED (JERRY_ES2015_BUILTIN_PROXY)
+  if (ECMA_OBJECT_IS_PROXY (binding_obj_p))
+  {
+    found_binding = ecma_proxy_object_has (binding_obj_p, name_p);
+
+    if (!ecma_is_value_true (found_binding))
+    {
+      return ECMA_IS_VALUE_ERROR (found_binding) ? found_binding : ECMA_VALUE_NOT_FOUND;
+    }
+  }
+  else
+  {
+#endif /* ENABLED (JERRY_ES2015_BUILTIN_PROXY) */
+    found_binding = ecma_op_object_find (binding_obj_p, name_p);
+
+    if (ECMA_IS_VALUE_ERROR (found_binding) || !ecma_is_value_found (found_binding))
+    {
+      return found_binding;
+    }
+
+#if ENABLED (JERRY_ES2015)
+    if (JERRY_LIKELY (lex_env_p == ecma_get_global_scope ()))
+#endif /* ENABLED (JERRY_ES2015) */
+    {
+      return found_binding;
+    }
+#if ENABLED (JERRY_ES2015_BUILTIN_PROXY)
+  }
```

Comment:
Note: This style of guarding is needed to eliminate the cpp-check error: `The scope of the variable 'found_binding' can be reduced` https://travis-ci.org/github/rerobika/jerryscript/jobs/685690005#L379